Assets/Scripts/HttpTopicClient.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.Networking;
+
+internal class Base64DecodedV1Token
+{
+    public string? api_key = null;
+    public string? endpoint = null;
+}
+
+public class HttpTopicSubscription
+{
+    private IEnumerator _poller;
+    private bool cancelled = false;
+
+    public bool Cancelled {
+        get { return cancelled; }
+    }
+
+    public void Cancel() {
+        Debug.Log("CANCELLING SUBSCRIPTION");
+        cancelled = true;
+    }
+
+    public HttpTopicSubscription() {}
+
+    public IEnumerator Poller
+    {
+        get { return _poller; }
+        set { _poller = value; }
+    }
+}
+
+public class HttpTopicClient
+{
+    private string endpoint;
+    private string authToken;
+
+    public HttpTopicClient(string authToken){
+        this.endpoint = GetEndpointFromKey(authToken);
+        Debug.Log("Using endpoint: " + this.endpoint);
+        this.authToken = authToken;
+    }
+
+    public void Publish(string cacheName, string topicName, string message)
+    {   
+        var contentType = "text/plain";
+        if (message.Contains("<sprite"))
+        {
+            Debug.Log("sending sprite message as binary");
+            contentType = "application/octet-stream";
+        }
+        var request = UnityWebRequest.Post(
+            "https://" + endpoint + "/topics/" + cacheName + "/" + topicName,
+            message,
```

Comment:
when you said in the PR description that you are exercising the binary path by posting an emoji, is it a unicode emoji? or is it the actual bytes from a gif/png?

I'm a tiny bit worried, looking at the docs for `UnityWebRequest.Post` here:

https://docs.unity3d.com/ScriptReference/Networking.UnityWebRequest.Post.html

The signature that you appear to be using (first signature in those docs) says this in the description:

> The data in postData will be interpreted into a byte stream via [System.Text.Encoding.UTF8](https://msdn.microsoft.com/en-us/library/system.text.encoding.utf8). The resulting byte stream will be stored in an [UploadHandlerRaw](https://docs.unity3d.com/ScriptReference/Networking.UploadHandlerRaw.html) and the Upload Handler will be attached to this UnityWebRequest.

If that means what it sounds like it means then it seems like you would have to pass an actual UTF-8 string here.

There are some other signatures a little later down in those docs that accept other arguments besides `string` but none of those look obviously correct either...

Reply:
Googling for this I keep seeing stuff like:

```
using (UnityWebRequest www = UnityWebRequest.Post(url, webform)) {
   www.uploadHandler = new UploadHandlerRaw(imageBytes);
   www.SendWebRequest();
}
```

Reply:
The actual content of the emojis in the system are plain text markup (i.e., `<sprite index="15">`). I just wanted an easily isolable place to switch messages to be published as binary, so I arbitrarily chose the emoji messages. It is 100% working to use the `Post` method in exactly the way I'm using it. Unity is doing the work to marshal the messages to []byte and is publishing exactly the payload we want depending on the content type I'm passing in. You can see in the console and using the SDKs that the correct message and encoding is being received by all the subscribers.